httpclient/errors.go

```diff
@@ -40,12 +42,12 @@ func DefaultErrorMapper(ctx context.Context, resp *http.Response, body io.ReadCl
 }
 
 func DefaultErrorRetriable(ctx context.Context, err error) bool {
-	switch some := err.(type) {
-	case *HttpError:
-		if some.StatusCode == 429 {
+	var httpError *HttpError
+	if errors.As(err, &httpError) {
+		if httpError.StatusCode == 429 {
 			return true
 		}
-		if some.StatusCode == 504 {
+		if httpError.StatusCode == 504 {
```

Comment:
QQ, not related to this change: isn't this supposed to be 503?

Reply:
504 is gateway timeout - https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/504, which is retriable. but we also need to retry on 503, yes.